core/src/main/scala/chisel3/experimental/trace.scala

```diff
@@ -0,0 +1,58 @@
+package chisel3.experimental
+
+import chisel3.internal.HasId
+import chisel3.{Aggregate, Data, Element, Module}
+import firrtl.AnnotationSeq
+import firrtl.annotations.{Annotation, CompleteTarget, SingleTargetAnnotation}
+import firrtl.transforms.DontTouchAnnotation
+
+/** The util that records the reference map from original [[Data]]/[[Module]] annotated in Chisel and final FIRRTL. */
+object trace {
+
+  /** Trace a Instance name. */
+  def traceName(x: Module): Unit = {
+    new ChiselAnnotation {
+      def toFirrtl: Annotation = TraceNameAnnotation(x.toTarget, x.toTarget)
+    }
+  }
+
+  /** Trace a Data name. */
+  def traceName(x: Data): Unit = {
+    x match {
+      case aggregate: Aggregate =>
+        annotate(new ChiselAnnotation {
+          def toFirrtl: Annotation = TraceNameAnnotation(aggregate.toAbsoluteTarget, aggregate.toAbsoluteTarget)
+        })
+        annotate(new ChiselAnnotation {
```

Comment:
You should probably use DontTouchAllTargets or the associated traits instead of emitting auxiliary dontTouch annotations.